### PR TITLE
DG Implemetation Edit and Help

### PR DESCRIPTION
**Pull request by chewwh09**
Help command
1. Add Intro to help feature
2. Add help sequence diagram and provide steps of the feature implementation
3. Add help activity diagram

Edit command
1. Add intro to edit feature
2. Add edit sequence diagram and provide steps of the feature implementation
3. Add edit activity diagram
 
Yet to complete design consideration for both features.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -147,7 +147,89 @@ Classes used by multiple components are in the `seedu.storemando.commons` packag
...
+Step 1: The user inputs `help` wanting to be redirected to the User Guide to learn more about StoreMando's command. <br>
+Step 2: The input is passed to `LogicManager` in a form of a String. <br>
+Step 3: The input string is passed to `StoreMandoParser` for it to be parsed. <br>
+Step 4: The input string is separated into command keyword and arguments containing the prefixes with the updated item's
+attribute. <br>
+Step 5: StoreMandoParser recognise that it is a `Help` command and creates a `HelpCommand` object. <br>
+Step 6: The `HelpCommand` is pass back to `LogicManager`. <br>
+Step 7: `LogicManager` then execute the `HelpCommand`. <br>
+Step 8: `HelpCommand` will create a `CommandResult` and pass to `LogicManager` with message stating 
+"Opened user guide information." <br>
```
Comment (JayChenYJ) [inline docs/DeveloperGuide.md]: do not put <br> in your code, that is against the coding standards for md files :)

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -147,7 +147,89 @@ Classes used by multiple components are in the `seedu.storemando.commons` packag
...
+
+#### Activity Diagram
+![HelpActivityDiagram](images/HelpActivityDiagram.png)
+
+#### Design Considerations
+- Alternative 1 (Current choice):
```
Comment (JayChenYJ) [inline docs/DeveloperGuide.md]: Design considerations for both features have to be added

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (chewwh09) [reply docs/DeveloperGuide.md]: All right noted!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (chewwh09) [reply docs/DeveloperGuide.md]: all right will do!

**Review by kumsssss — COMMENTED** [review]
LGTM!

**Review by Md-Fazil — COMMENTED** [review]
Lgtm!

**Review by JayChenYJ — APPROVED** [review]

**Pull request merged**